ephios/user_management/models.py

```diff
@@ -189,3 +189,68 @@ class QualificationGrant(Model):
 
     def __str__(self):
         return f"{self.qualification!s}, {self.user!s}"
+
+
+class Consequence(Model):
+    slug = models.CharField(max_length=255)
+    data = models.JSONField(default=dict)
+
+    user = models.ForeignKey(
+        get_user_model(), on_delete=models.CASCADE, verbose_name=_("affected user"), null=True
+    )
+    shift = models.ForeignKey(
```

Comment:
why would we want the shift? the qualification-consequence doesn't need it and the workhour-consequences saves the participation to the data dict

Reply:
still not sure about this

Reply:
yeah so after checking this makes no sense for working hours. so do we need it here or can we move it to the JSON? if it needs to stay here we need a blank=True for django admin

Reply:
It's currently used to determine who can confirm consequences...

Reply:
So after talking about this, there are two things to consider when deciding between the json field and a foreign key:

* prefetching objects: When rendering a list of consequences, needed objects should be prefetchable using the IDs stored in json. From what I read and tried: `prefetch_related` and `select_related` only work with actual foreign keys and don't support a pk stored in json. You could probably build a query using `extra`, but that's another topic.

* filtering consequences: you should be able to filter consequences based on data in json. That's possible with modern django.

Reply:
I went for annotations for now. Bulks up the code quite a bit :/